### PR TITLE
Add setting to only display control characters

### LINKED ISSUE #66674: Screencast mode - settings for what to show
It would be nice to have control in settings to show non character (control keys) only vs show all. For example, showing CMD, CTRL, SHIFT, ESC, ENTER only vs showing all a, b, c, d .... z etc.

### PR DESCRIPTION
**Pull request by mkenigs**
Fixes #66674 
Creates settings for Screencast Mode and a setting to only display control characters, and changes Screencast Action accordingly.

### CODE REVIEW

**Comment by jbreuer95** [comment]
+1

**Comment by texastoland** [comment]
Excited to try this! Does it handle multi-key shortcuts like Markdown preview too?

**Comment by mkenigs** [comment]
It works for multi-key shortcuts, though screencast mode as a whole doesn't handle chords yet.

**Review by texastoland — COMMENTED** [review]
CC @mkenigs 👀

File: src/vs/workbench/electron-browser/actions/developerActions.ts [hunk 1]
```diff
@@ -193,13 +196,15 @@ export class ToggleScreencastModeAction extends Action {
 			const label = keybinding.getLabel();
 
 			if (!event.ctrlKey && !event.altKey && !event.metaKey && !event.shiftKey && this.keybindingService.mightProducePrintableCharacter(event) && label) {
-				keyboardMarker.textContent += ' ' + label;
+				if (!this.configurationService.getValue<boolean>('screencastMode.onlyControlKeys')) {
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply src/vs/workbench/electron-browser/actions/developerActions.ts]: Assuming screencastMode.onlyControlKeys is true, if I change it back, an empty bar will show up across the screen when I type a non-modifier key. The unwritten else is to pass, leaving keyboardMarker.style.display = 'none' as it's set above

File: src/vs/workbench/electron-browser/main.contribution.ts [hunk 2]
```diff
@@ -692,4 +692,18 @@ import { LogStorageAction } from 'vs/platform/storage/node/storageService';
...
+
+	// Screencast
+	registry.registerConfiguration({
+		'id': 'screencastMode',
+		'title': nls.localize('screencastModeConfigurationTitle', "Screencast Mode"),
+		'type': 'object',
+		'properties': {
+			'screencastMode.onlyControlKeys': {
+				'type': 'boolean',
+				'description': nls.localize('screencastMode.onlyControlKeys', "Only show control keys in Screencast Mode."),
```
[the remark this answers is not in the document]
Comment (mkenigs) [reply src/vs/workbench/electron-browser/main.contribution.ts]: Agreed, pushed a commit! Honestly would probably be better to tweak this so it was just all keyboard shortcuts - thoughts?

[the remark this answers is not in the document]
Comment (texastoland) [reply src/vs/workbench/electron-browser/main.contribution.ts]: That does make sense. The only reason I could think not to is backwards incompatibility. I think that's a nongoal here.

File: src/vs/workbench/electron-browser/actions/developerActions.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (texastoland) [reply src/vs/workbench/electron-browser/actions/developerActions.ts]: This looks close CC @mjbvz. Could we get an additional reviewer?

**Comment by mkenigs** [comment]
@texastoland wish I could have gotten to this sooner but anyways, I changed the setting to whether something is a keyboard shortcut, not just a modifier key. Seems to me like that's what's going to be desired most of the time. Let me know what you think

**Comment by mkenigs** [comment]
Bonus feature: makes it super easy to add support for chords

**Comment by joaomoreno** [comment]
@mkenigs You seem to do a lot more here than expected:

- control characters
- chords
- setting
- setting category

Usually we prefer to have everything in separate PRs, but let's take it from here.

---

The intended feature for this PR works fine. But not chord handling.

Chords work in the following key sequence:

- press ⌘
- press K, release K
- press C, release C
- release ⌘

But they don't work in the following:

- press ⌘
- press K, release K
- release ⌘
- press ⌘
- press C, release C
- release ⌘

**Comment by mkenigs** [comment]
@joaomoreno thanks for looking at this! Changed it so that screencast mode only ever displays keys that have dispatch parts. That fixes the issue with chords, and it also allows slightly simpler logic regarding keys like shift.

Chords still won't display properly if the screencast mode timeout is reached before that of chords - should those be changed to match?

**Comment by mkenigs** [comment]
@joaomoreno Anything else I can do on this?

**Comment by joaomoreno** [comment]
Cleaned up a bit, removed the chord handling. Thanks! :beers:

**Pull request merged**